File: LeetCodeExample.Test/Binary Tree/_00102_BinaryTreeLevelOrderTraversal.py
# Given the root of a binary tree, return the level order traversal of its nodes' values. (i.e., from left to right, level by level).

# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
from typing import Optional, List
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        self.ans = []
        def rec(node: Optional[TreeNode], level: int):
            if not node:
                return
            if level > len(self.ans) - 1:
                self.ans.append([])
            self.ans[level].append(node.val)
            rec(node.left, level + 1)
            rec(node.right, level + 1)
        if not root:
            return []
        rec(root, 0)
        return self.ans

File: LeetCodeExample.Test/Binary Tree/_00102_BinaryTreeLevelOrderTraversal.cs
using LeetCodeExample.Test.Common;
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Linq;

namespace LeetCodeExample.Test;

/// <summary>
//Given the root of a binary tree, return the level order traversal of its nodes' values. (i.e., from left to right, level by level).
/// </summary>
public class _00102_BinaryTreeLevelOrderTraversal
{
    public IList<IList<int>> LevelOrder(TreeNode root)
    {
        List<IList<int>> rtn = new();
        if (root == null)
            return rtn;
        Queue<TreeNode> q = new();
        q.Enqueue(root);
        while (q.Count != 0)
        {
            List<int> row = new();
            rtn.Add(row);
            int count = q.Count;
            for (int i = 0; i < count; i++)
            {
                TreeNode node = q.Dequeue();
                row.Add(node.val);
                if (node.left != null)
                    q.Enqueue(node.left);
                if (node.right != null)
                    q.Enqueue(node.right);
            }
        }
        return rtn;
    }
}